File: mooc-2024/part-4/4-define-iteration/08-the-sum-of-lists/main.c
#include <stdio.h>
#include "../../../../lib/input.h"

int main() {
    const int len = 5;
    int l1[] = {1,2,3,4,5};
    int l2[] = {1,2,3,4,5};
    int res[len];

    for (int idx = 0; idx < len; idx++) {
        res[idx] = l1[idx] + l2[idx];
    }

    print_int_arr(res, len);
    printf("\n");
    
    return 0;
}

File: mooc-2024/part-4/4-define-iteration/08-the-sum-of-lists/main.py
def list_sum(l1: list, l2: list):
    res = [None] * len(l1)
    for idx in range(len(l1)):
        res[idx] = l1[idx] + l2[idx]
    return res

a = [1, 2, 3]
b = [7, 8, 9]
print(list_sum(a, b)) # [8, 10, 12]